src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1236,7 +1241,144 @@ subroutine eigencalving(domain, err)
+      !config_default_flowParamA) ! REMOVE THIS ONCE YOU CAN GET A FROM
+      !ALBANY!!!!!
+      call mpas_pool_get_config(liConfigs, 'config_flowLawExponent', config_flowLawExponent)
+      call mpas_pool_get_config(liConfigs, 'config_use_Albany_flowA_eqn', config_use_Albany_flowA_eqn)
+
+      block => domain % blocklist
+
+      do while (associated(block))
+         ! get pools
+          call mpas_pool_get_subpool(block % structs, 'geometry', geometryPool)
+          call mpas_pool_get_subpool(block % structs, 'velocity', velocityPool)
+          call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
+          call mpas_pool_get_subpool(block % structs, 'scratch', scratchPool)
+          call mpas_pool_get_subpool(block % structs, 'thermal', thermalPool)
+         ! get fields
+          call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+          call mpas_pool_get_array(geometryPool, 'cellMask', cellMask)
+          call mpas_pool_get_array(geometryPool, 'layerThickness', layerThickness)
+          call mpas_pool_get_array(velocityPool, 'eMax', eMax)
+          call mpas_pool_get_array(velocityPool, 'eMin', eMin)
+          call mpas_pool_get_array(velocityPool, 'vonMisesStress', vonMisesStress)
+          call mpas_pool_get_array(velocityPool, 'flowParamA', flowParamA)
+          call mpas_pool_get_array(velocityPool, 'xvelmean', xvelmean)
+          call mpas_pool_get_array(velocityPool, 'yvelmean', yvelmean)
+          call mpas_pool_get_array(geometryPool, 'calvingThickness', calvingThickness)
+          call mpas_pool_get_array(geometryPool, 'calvingVelocity', calvingVelocity) 
+          call mpas_pool_get_array(geometryPool, 'thickness', thickness)
+          call mpas_pool_get_array(thermalPool, 'temperature', temperature)
+          
+          vonMisesStress(:) = 0.0_RKIND
+
+          ! get flowParamA from MPAS or use Albany-like equation
+          if ( config_use_Albany_flowA_eqn ) then
+          !calculate Albany-type flowParamA
+          else
+             call li_calculate_flowParamA(meshPool, temperature, thickness,flowParamA,err) ! Get MPAS flowParamA
+          endif
+    
+         !Using a depth-averaged ice viscosity parameter B_depthAvg
+         !=sum(layerThickness(:,iCell) *
+         !flowParamA(:,iCell)**(-1.0_RKIND/config_flowLawExponent), dim=1) /
+         !thickness(iCell)
+         ! Calculate effective von  Mises stress.
+          calvingVelocity(:) = 0.0_RKIND
+
+          do iCell = 1,nCells
+             if ( thickness(iCell) > 0.0_RKIND ) then
+                vonMisesStress(iCell) = sqrt(3.0_RKIND) * sum(layerThickness(:,iCell) * &
+                              flowParamA(:,iCell)**(-1.0_RKIND/config_flowLawExponent),dim=1) / &
+                              thickness(iCell) * ( 0.5_RKIND * ( (max(0.0_RKIND, eMax(iCell)))**2 + &
+                              (max(0.0_RKIND, eMin(iCell)))**2) )**(1 / (2 * config_flowLawExponent))
```

Comment:
Make the exponent integers into floating point _RKIND types (and check if it changes your answers - I don't think it will because I believe there will be automatic typecasting going on already).

Reply:
Currently running this test.

Reply:
I've confirmed that vonMisesStress and calvingVelocity are BFB with integers and floating point _RKIND types here.

Reply:
Ok, thanks for checking that out and for keeping the explicit floating point version.